callback.php

```diff
@@ -7,10 +7,9 @@
 
 /* Exchange the authorisation code for an access_token */
 $response = $sageone_client->getAccessToken($_GET['code']);
-$access_token = json_decode($response, true)['access_token'];
 
-/* redirect with the access_token */
-header("Location: http://localhost:8000/sageone_data.php?access_token=" . $access_token);
+/* redirect with the response */
+header("Location: http://localhost:8000/sageone_data.php?token_response=" . $response);
```

Comment:
You really should urlencode `$response`, or is `$sageone_client->getAccessToken($_GET['code'])` doing that?